### DISCUSSION TITLE
关于 skynet.pack 对于打包后数据的处理。

### DISCUSSION THREAD

**Discussion by Gowa2017** [cloudwu/skynet#1498] [General]
```lua
  local i      = 10004
  local is, sz = skynet.pack(string.pack(">I2", i))
  local iss    = skynet.tostring(is, sz)
  print(iss:byte(1) * 256 + iss:byte(2))
  print(iss:byte(2) * 256 + iss:byte(1))
```
输出：
```
5159
10004
```

这段代码，我期待我的 i 是经大端字节序打包后，再 pack，但最终和我相反，tostring 后，用小端才能解析出来。
场景是：我在一个单独的服务将协议ID大端打包后，发送到另外一个服务，这个服务负责在协议数据前后加上校验等等。
但拿到的协议ID不对。

**Comment by cloudwu** [comment]
读 wiki 上的文档，搞清楚 skynet.pack 和 skynet.tostring 在做什么。

如果找不到文档或读不懂文档，可以看代码的实现弄清楚。

**Reply by Gowa2017** [reply]
抱歉，看了 一下代码。skynet.pack 加上了类型信息。得用 unpack 才能拿出来。
我的场景，直接返回 Lua 字符串即可，或用packstring.
